setup.py

```diff
@@ -1,26 +1,18 @@
 #!/usr/bin/env python
 
+import glob
 import sys
 import os
 import warnings
 
+from setuptools import setup, Extension
 from distutils.dist import Distribution
 
 display_option_names = Distribution.display_option_names + ['help', 'help-commands']
 query_only = any('--' + opt in sys.argv for opt in display_option_names) or len(sys.argv) < 2 or sys.argv[1] == 'egg_info'
 
-# Use setuptools for querying the package, normal builds use distutils
```

Comment:
It breaks on `python setup.py sdist` if using `distutils` here in my environment:

```
$ python setup.py sdist
setup.py:81: UserWarning: Cannot find ta-lib library, will try to build from source.
  'Cannot find ta-lib library, will try to build from source.'
/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'setup_requires'
  warnings.warn(msg)
/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
running sdist
running check
error: [Errno 2] No such file or directory: 'venv/lib/python2.7/lib-dynload/lib-dynload'
```

@mrjbq7 Would you please detail why the following code block for `setup` is essential here?